ExampleMod/Content/Tiles/ExampleVanillaConversionTiles.cs

```diff
@@ -38,66 +41,72 @@ public bool ConvertToHallow(int i, int j, int type, int conversionType) {
-				WorldGen.ConvertTile(i, j, TileID.DesertFossil);
-				return false;
+			switch (conversionType) {
+				// Purification powder doesn't convert hallow tiles back into purity, so we don't check for BiomeCoversionID.PurificationPowder
+				case BiomeConversionID.Purity:
+				case BiomeConversionID.Sand: // Yellow (desert) solution also converts evil/hallowed tiles back into purity, so don't forget that check!
+					WorldGen.ConvertTile(i, j, TileID.DesertFossil);
+					return false;
+				case BiomeConversionID.Corruption:
+					WorldGen.ConvertTile(i, j, ModContent.TileType<CorruptFossilTile>());
+					return false;
+				case BiomeConversionID.Crimson:
+					WorldGen.ConvertTile(i, j, ModContent.TileType<CrimsonFossilTile>());
+					return false;
+
+				// This example showcases how to make hallow and evil biome conversion work, but you can extend this code to work for the other vanilla solutions.
+				// Just don't forget to register the conversion type in SetStaticDefaults if you want a vanilla tile to turn into your new modded tile.
+				// case BiomeConversionID.Snow:
+				//		WorldGen.ConvertTile(i, j, TileID.Slush);
+				//		return false;
 			}
-			else if (conversionType == BiomeConversionID.Corruption) {
-				WorldGen.ConvertTile(i, j, ModContent.TileType<CorruptFossilTile>());
-				return false;
-			}
-			else if(conversionType == BiomeConversionID.Crimson) {
-				WorldGen.ConvertTile(i, j, ModContent.TileType<CrimsonFossilTile>());
-				return false;
-			}
-			//This example showcases how to make hallow and evil biome conversion work, but you can extend this code to work for the other vanilla solutions.
-			//Just don't forget to register the conversion type in SetStaticDefaults if you want a vanilla tile to turn into your new modded tile.
-			//else if (conversionType == BiomeConversionID.Snow) {
-			//	WorldGen.ConvertTile(i, j, TileID.Slush);
-			//}
 
 			return true;
 		}
 
 		public override void RandomUpdate(int i, int j) {
-			//Account for journey mode disabling infections and plantera slowing down biome spread rate
-			if (Main.hardMode && (NPC.downedPlantBoss && WorldGen.genRand.NextBool(2)) || !WorldGen.AllowedToSpreadInfections)
+			// Don't spread the biome if disabled by journey mode
+			if (!WorldGen.AllowedToSpreadInfections)
+				return;
+			// Spreading tiles (besides grass) don't spread until hardmode. After plantera, the spread rate is also reduced, so we have to account for that
+			if (!Main.hardMode || (NPC.downedPlantBoss && WorldGen.genRand.NextBool(2)))
 				return;
 
 			//Check a random nearby tile and try to convert it into hallow
 			int testX = i + WorldGen.genRand.Next(-3, 4);
 			int testY = j + WorldGen.genRand.Next(-3, 4);
 			if (!WorldGen.InWorld(testX, testY, 10))
-				continue;
+				return;
 
 			//Chlorophyte prevents and repels biome spread, so we should account for that
 			if (WorldGen.nearbyChlorophyte(testX, testY)) {
 				WorldGen.ChlorophyteDefense(testX, testY);
```

Comment:
Recommend adding a `return` after `ChlorophyteDefense`. Also please add docs to `ChlorophyteDefense`